In [21]:
import pandas as pd

In [22]:
df = pd.read_csv('czech-financial-dataset/data/loan.csv')
df.head()

loan_id  account_id    date  amount  duration  payments status
0     5314        1787  930705   96396        12    8033.0      B
1     5316        1801  930711  165960        36    4610.0      A
2     6863        9188  930728  127080        60    2118.0      A
3     5325        1843  930803  105804        36    2939.0      A
4     7240       11013  930906  274740        60    4579.0      A

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 682 entries, 0 to 681
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   loan_id     682 non-null    int64  
 1   account_id  682 non-null    int64  
 2   date        682 non-null    int64  
 3   amount      682 non-null    int64  
 4   duration    682 non-null    int64  
 5   payments    682 non-null    float64
 6   status      682 non-null    object 
dtypes: float64(1), int64(5), object(1)
memory usage: 37.4+ KB


In [24]:
df['account_id'].unique()

array([ 1787,  1801,  9188,  1843, 11013,  8261, 11265,  5428, 10973,
        4894, 10364,  5270,  6034,  6701,  2705,  7123, 10320,  9034,
        9236, 10266,     2,  7401,  2486,  1071,  8475, 11111,  1603,
        1166,  8073, 10940,  5313, 11135,  2694,  7753,  7445,  7485,
        8330,  6555, 10079,  5385,  5891, 10131,  8321,  5256,   813,
        2933,  8173,  9833,  7769,  6473,  7454,  6930,   173,  8051,
       10439,  9337,  4576,  8027,  5927,  6273, 11065,  8558,  5442,
        1811,  7844,  4260,  3354,  2762,  6051,  9265,  5228,  2824,
         426,  4937,   790, 10911,  8129,  2176,  9093,  5215,  9403,
        9104,  5120,  8519, 10788,  8260,  4259, 10105,  3050,  3674,
        5481,  9307,  4427,  3834,  7824,  8566,  7965,  5952,  6040,
        4448,  8982,  6363,  1012,  2167,  5622,  5991,   290,  6148,
        7133,  8041,  1106,  1527,  1978,  4337,  2345, 10451,  6425,
        6226,   344,  2962, 10738,  3747,  7523,  3166,  6118,   816,
        9640,  6715,

In [25]:
df = pd.read_csv('czech-financial-dataset/data/disp.csv')
df.head()

disp_id  client_id  account_id       type
0        1          1           1      OWNER
1        2          2           2      OWNER
2        3          3           2  DISPONENT
3        4          4           3      OWNER
4        5          5           3  DISPONENT

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5369 entries, 0 to 5368
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   disp_id     5369 non-null   int64 
 1   client_id   5369 non-null   int64 
 2   account_id  5369 non-null   int64 
 3   type        5369 non-null   object
dtypes: int64(3), object(1)
memory usage: 167.9+ KB


In [27]:
df.isna().sum()

disp_id       0
client_id     0
account_id    0
type          0
dtype: int64

In [28]:
for column in df.columns:
    print(column)
    print(df[column].unique())

disp_id
[    1     2     3 ... 13660 13663 13690]
client_id
[    1     2     3 ... 13968 13971 13998]
account_id
[    1     2     3 ... 11359 11362 11382]
type
['OWNER' 'DISPONENT']


Our "13998" clients are using "11382" account with the total of two types: owner or disponent

-meaning =  client_id will be use as a base column

In [29]:
#changing object column into numerical values

mapping = {'OWNER': 1, 'DISPONENT': 0}
df['AccStat'] = df['type'].replace(mapping)

In [30]:
df.head()

disp_id  client_id  account_id       type  AccStat
0        1          1           1      OWNER        1
1        2          2           2      OWNER        1
2        3          3           2  DISPONENT        0
3        4          4           3      OWNER        1
4        5          5           3  DISPONENT        0

In [49]:
df = pd.read_csv('czech-financial-dataset/data/trans.csv')
df.head()

C:\Users\hazir\AppData\Local\Temp\ipykernel_11744\4025008582.py:1: DtypeWarning: Columns (8) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('czech-financial-dataset/data/trans.csv')


trans_id  account_id    date    type operation  amount  balance k_symbol  \
0    695247        2378  930101  PRIJEM     VKLAD   700.0    700.0      NaN   
1    171812         576  930101  PRIJEM     VKLAD   900.0    900.0      NaN   
2    207264         704  930101  PRIJEM     VKLAD  1000.0   1000.0      NaN   
3   1117247        3818  930101  PRIJEM     VKLAD   600.0    600.0      NaN   
4    579373        1972  930102  PRIJEM     VKLAD   400.0    400.0      NaN   

  bank  account  
0  NaN      NaN  
1  NaN      NaN  
2  NaN      NaN  
3  NaN      NaN  
4  NaN      NaN

In [50]:
#drop columns
drop_cols = [
    'k_symbol',
    'bank',
    'account'
]
df = df.drop(drop_cols, axis=1)

#remove missing values
df.dropna(subset=['operation'], inplace=True)

#rename columns
df = df.rename(columns={'type': 'TransType', 'operation': 'TransOp','date':'TransDate'})

#Change the czech terms into english
mapping = {'PRIJEM':'INCOME', 'VYDAJ':'EXPENSE', 'VYBER':'WITHDRAWAL'}
mapping2 = {'VKLAD': 'depositing money', 'PREVOD Z UCTU':'transfer from account', 'VYBER':'withdrawing money', 
            'PREVOD NA UCET':'transfer to account','VYBER KARTOU':'withdrawing by card'}
df['TransType'] = df['TransType'].replace(mapping)
df['TransOp'] = df['TransOp'].replace(mapping2)

#Changing the date column into DateTime Object
df['TransDate'] = pd.to_datetime(df['TransDate'], format='%y%m%d')

In [51]:
df.head()

trans_id  account_id  TransDate TransType           TransOp  amount  \
0    695247        2378 1993-01-01    INCOME  depositing money   700.0   
1    171812         576 1993-01-01    INCOME  depositing money   900.0   
2    207264         704 1993-01-01    INCOME  depositing money  1000.0   
3   1117247        3818 1993-01-01    INCOME  depositing money   600.0   
4    579373        1972 1993-01-02    INCOME  depositing money   400.0   

   balance  
0    700.0  
1    900.0  
2   1000.0  
3    600.0  
4    400.0

In [52]:
df['TransType'].unique()

array(['INCOME', 'EXPENSE', 'WITHDRAWAL'], dtype=object)

In [53]:
data = []
for account in df['account_id'].unique():
    x = df[df['account_id'] == account]
    income = x[x['TransType'] == 'INCOME']['amount'].sum()
    expense = x[x['TransType'] == 'EXPENSE']['amount'].sum()
    withdraw = x[x['TransType'] == 'WITHDRAWAL']['amount'].sum()
    d = {'AccId': account, 'IncomeSum': income , 'ExpenseSum': expense, 'WithdrawSum': withdraw}
    data.append(d)

df = pd.DataFrame(data=data)
df.head()

AccId  IncomeSum  ExpenseSum  WithdrawSum
0   2378  2690516.0   2433574.0     177736.0
1    576   447804.0    420034.2          0.0
2    704  1367112.0   1296604.2          0.0
3   3818  1394648.0   1345498.2          0.0
4   1972   381856.0    364406.2          0.0

KeyboardInterrupt: 